java/memory/memory-core/src/main/java/org/apache/arrow/memory/BufferLedger.java

```diff
@@ -364,8 +364,12 @@ boolean transferBalance(final ReferenceManager targetReferenceManager) {
             targetReferenceManager.getAllocator().getName());
       }
 
-      boolean overlimit = targetAllocator.forceAllocate(allocationManager.getSize());
-      allocator.releaseBytes(allocationManager.getSize());
+      long size = allocationManager.getSize();
+      targetAllocator.getListener().onPreAllocation(size);
+      boolean overlimit = targetAllocator.forceAllocate(size);
+      targetAllocator.getListener().onAllocation(size);
+      allocator.releaseBytes(size);
+      allocator.getListener().onRelease(size);
```

Comment:
If we do not validate "assertEquals(1, l1.getNumReleaseCalls())" or "assertEquals(1, l2.getNumCalls())", question: What could be the errors in current programs to this changes should be needed ?

Reply:
For example. Buffer1 and Buffer2 are shared and were created by Allocator1 and Allocator2. Allocator1 has a listener assigned but Allocator2 doesn't have one. Once the memory ownership is transferred from Buffer1 to Buffer2, then the listener will not be aware of that transferring, and `listener.free()` will never be called even we close both Buffer1 and Buffer2.

Reply:
Ok thank you for clarification